### PR TITLE
Modify rule S3453: LaYC format

### PR DESCRIPTION
**Pull request by sebastien-marichal**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S3453/why-dotnet.adoc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+== Why is this an issue?
+
+When a class has only a `private` constructor, it can't be instantiated except within the class itself.
+Such classes can be considered https://en.wikipedia.org/wiki/Dead_code[dead code] and should be fixed
+
+=== Exceptions
+
+* Classes that access their private constructors (singletons or smart enums) are ignored.
```
Comment (gregory-paidis-sonarsource) [inline rules/S3453/why-dotnet.adoc:8]: I would link singleton to [this chapter](https://csharpindepth.com/articles/singleton) to explain the pattern. Same in documentation

**Review by gregory-paidis-sonarsource — CHANGES_REQUESTED** [review]
Very good! Left a minor suggestion :)

File: rules/S3453/why-dotnet.adoc (hunk 1 above)
Comment (sebastien-marichal) [reply rules/S3453/why-dotnet.adoc]: I added a link to both singletons and smart enums.
I linked the singleton one to the wiki page to be language agnostics and added your suggestion to the `Articles & blog posts` section.
For the smart enum, I was not able to find an appropriate link for VB so I link the documentation about enumeration classes, it is for C# but I think it is still good for VB.NET as well.
Let me know what you think.

Comment (gregory-paidis-sonarsource) [reply rules/S3453/why-dotnet.adoc]: Seems fine, good job!

**Review by gregory-paidis-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**